CMakeLists.txt

```diff
@@ -145,11 +145,16 @@ add_dependencies(slam_toolbox ${PROJECT_NAME}_generate_messages_cpp)
 target_link_libraries(slam_toolbox ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 
 #### Merging maps tool
-add_executable(merge_maps_tool src/merge_maps_tool.cpp)
-add_dependencies(merge_maps_tool ${PROJECT_NAME}_generate_messages_cpp)
-target_link_libraries(merge_maps_tool ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+add_executable(merge_maps_kinetic src/merge_maps_kinetic.cpp)
+add_dependencies(merge_maps_kinetic ${PROJECT_NAME}_generate_messages_cpp)
+target_link_libraries(merge_maps_kinetic ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 
-install(TARGETS slam_toolbox merge_maps_tool spa_solver_plugin g2O_solver_plugin ceres_solver_plugin SlamToolboxPlugin
+#### Merging maps optimizaiton
+add_executable(merge_maps_optimization src/merge_maps_optimization.cpp)
```

Comment:
Please take out all your merge maps optimization work out of this PR unless its complete. Meta-PRs are impossible to review